compress/gzip.go

```diff
@@ -7,15 +7,26 @@ import (
 	"compress/gzip"
 	"github.com/xitongsys/parquet-go/parquet"
 	"io/ioutil"
+	"sync"
 )
 
+var gzipWriterPool sync.Pool
+
 func init() {
+	gzipWriterPool = sync.Pool{
+		New: func() interface{} {
+			return gzip.NewWriter(nil)
+		},
+	}
+
 	compressors[parquet.CompressionCodec_GZIP] = &Compressor{
 		Compress: func(buf []byte) []byte {
-			var res bytes.Buffer
-			gzipWriter := gzip.NewWriter(&res)
+			res := new(bytes.Buffer)
+			gzipWriter := gzipWriterPool.Get().(*gzip.Writer)
+			gzipWriter.Reset(res)
 			gzipWriter.Write(buf)
 			gzipWriter.Close()
+			gzipWriterPool.Put(gzipWriter)
```

Comment:
Add `gzipWriter.Reset(nil)` this will remove the reference to the buffer when it is in the pool.

Reply:
Thank you very much, fixed !